Mention `[]` operator also being able to thrown a runtime exception with indexers (#41655)
dotnet/docs: docs/csharp/nullable-references.md

When you dereference a variable whose value is `null`, the runtime throws a <xref:System.NullReferenceException?displayProperty=nameWithType>.

> Similarly warnings can be produced when `[]` notation is used on an indexer whose values were not initialized:

Reviewer: I'm concerned that this text reads like the *collection* is null, rather than the *object*. How about this:

```suggestion
Similarly warnings can be produced when `[]` notation to access a member of an object when the object is `null`:
```

Author: Yes, `values were not initialized` indeed sounds like `[null, null, null]`, while the whole `c` will be `null`.
Your suggestion should be applied, with the only concern whether there's no `is used` missing by any chance here?
> be produced when `[]` notation **is used** to access a member of

Reviewer: Thanks @BartoszKlonowski 

I updated my comment. I'll accept it and merge this PR.
